src/WebJobs.Script.WebHost/Health/TokenExpirationService.cs

```diff
@@ -43,19 +43,26 @@ public TokenExpirationService(IEnvironment environment, ILogger<TokenExpirationS
 
         protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         {
-            await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
-            await _standby.Task.WaitAsync(cancellationToken);
-
-            if (_environment.IsCoreTools())
+            try
             {
-                return;
-            }
+                await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
+                await _standby.Task.WaitAsync(cancellationToken);
 
-            while (!cancellationToken.IsCancellationRequested)
+                if (_environment.IsCoreTools())
+                {
+                    return;
+                }
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    // check at every hour
+                    AnalyzeSasTokenInUri();
+                    await Task.Delay(TimeSpan.FromHours(1), cancellationToken);
+                }
+            }
+            catch (Exception ex) when (ex.GetType() != typeof(OperationCanceledException))
```

Comment:
nit -- you can write this a little cleaner with `(ex is not OperationCanceledException)`